packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.test.tsx

```diff
@@ -35,32 +34,40 @@ describe('ApplicationExtension', () => {
         })
     })
 
-    describe('extendRoutes', () => {
-        test('should return the routes array without modification', () => {
-            const routes: RouteProps[] = [
-                {path: '/home', component: mockComponent},
-                {path: '/about', component: mockComponent}
-            ]
-
-            const result = extension.extendRoutes(routes)
-            expect(result).toEqual(routes)
+    describe('getRoutes', () => {
+        test('initially return an empty array', () => {
+            const routes = extension.getRoutes({locals: {}})
+            expect(routes).toHaveLength(0)
         })
 
-        test('should allow for modification of routes', () => {
-            const routes: RouteProps[] = [{path: '/home', component: mockComponent}]
+        test('should allow adding a new route', () => {
             const additionalRoute: RouteProps = {path: '/new', component: mockComponent}
+            const getRoutesSpy = jest.spyOn(extension, 'getRoutes').mockImplementation(() => {
+                return [additionalRoute]
+            })
+
+            const routes = extension.getRoutes({locals: {}})
+            expect(routes).toContainEqual(additionalRoute)
+            expect(routes).toHaveLength(1)
 
-            const extendRoutesSpy = jest
-                .spyOn(extension, 'extendRoutes')
-                .mockImplementation((baseRoutes) => {
-                    return [...baseRoutes, additionalRoute]
-                })
+            getRoutesSpy.mockRestore()
+        })
+    })
+
+    describe('getRoutesAsync', () => {
+        test('initially undefined', () => {
+            // eslint-disable-next-line @typescript-eslint/unbound-method
+            expect(extension.getRoutesAsync).toBeUndefined()
+        })
+    })
```

Comment:
What are we testing here?

Reply:
I'll rename the test (and maybe tweak it) to make it clearer. I wanted to make sure that `getRoutesAsync` is initially not implemented.